File: compute.py
#!/usr/bin/python

import sys
import math

file=open(sys.argv[1],'r')
num=0
total=0
for line in file.xreadlines():
	element=line[:-1].split('\t')
	total=total+float(element[0])*float(element[1])
	num=num+float(element[0])	
mean=total/num
print total 
print num
print "mean "+str(mean)
file=open(sys.argv[1],'r')
std=0
csize=float(0)
median=0
for line in file.xreadlines():
	element=line[:-1].split('\t')
	if csize < num/float(2) and num/float(2) <= csize+float(element[0]):
		median=element[1]
	std=std+(float(element[1])-mean)*float(element[0])
	csize=csize+float(element[0])
print std
print num
print "std "+str(math.sqrt(std/num))
print "median "+median

File: compute.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
 
#define MAXCHAR 1000

typedef unsigned short int Boolean;

#define TRUE 1
#define FALSE 0

int get_length(char* filname); // to compute the size of file

int main(int argc, char *argv[]) {
	FILE *fp;
	char str[MAXCHAR];
	char* filename = argv[1];
	float size = (float) get_length(filename);
	fp = fopen(filename, "r");
	float middle = size/2;
	int count = 1;
	float m = 0;
	float std = 0;
	float med = 0;
	int p_value = 0;
	int sum = 0;

	Boolean first = TRUE;

	if (fp == NULL) {
		printf("Could not open file %s",filename);
		return 1;
	}
	// loop over each line of the file
	while (fgets(str, MAXCHAR, fp) != NULL) {
		// printf("%s", str);
		// compute Mean and SD
		int value = atoi(str);
		sum += value;
		if (first == TRUE) {
			m = value;
			first = FALSE;
		}
		else {
			float u_m = m + ((float) value-m)/((float) count);
			std = std + (value-m)*(value-u_m);
			m = u_m;
		}
		// compute Median
		if (count-1 < middle && middle < count)
			med = value;	
		else if (count == middle)
			p_value = value;
		else if (count == middle+1)
			med = (p_value+value)/2;
		count++;
	}
	printf("Mean: %f\n", (float)sum/(float)size);
	printf("SD: %f\n", sqrt(std/(size-1)));
	printf("Median: %f\n", med);
	fclose(fp);
	return 0;
}

int get_length(char* filename) {
	FILE *fp;
        char str[MAXCHAR];
	int count = 0;
	fp = fopen(filename, "r");
	
	if (fp == NULL) {
                printf("Could not open file %s",filename);
                return 1;
        }
        while (fgets(str, MAXCHAR, fp) != NULL) {
                count++;
        }
	fclose(fp);
	return count;
}
